ts/components/session/SessionIconButton.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import classNames from 'classnames';
+import { exists } from 'fs';
+
+export enum SessionIconButtonTypes {
+  'exit' = 'exit',
+  'search' = 'search',
+  'back' = 'back',
+  'attachment' = 'attachment',
+  'emoji' = 'emoji',
+  'favorite' = 'favorite',
+  'group' = 'group',
+  'menu' = 'menu',
+  'message' = 'message',
+  'microphone' = 'microphone',
+  'network' = 'network',
+  'options' = 'options',
+  'theme' = 'theme',
+}
+
+export enum SessionIconButtonSizes {
+  'small' = 'small',
+  'medium' = 'medium',
+  'large' = 'large',
+}
+
+interface Props {
+  iconType: SessionIconButtonTypes;
+  iconSize: SessionIconButtonSizes;
+  onClick: any;
+}
+
+export class SessionIconButton extends React.PureComponent<Props> {
+  constructor(props: any) {
+    super(props);
+    this.clickHandler = this.clickHandler.bind(this);
+  }
+
+  public render() {
+    const { iconType, iconSize } = this.props;
+
+    const iconPath = `./images/session/icon-${iconType}.svg`;
+
+    return (
+      <div
+        className={classNames(
+          'session-icon-button',
+          iconType === SessionIconButtonTypes.exit ? 'exit' : '',
```

Comment:
Rather than doing this, you can just pass the value of `iconType` straight into the function

Reply:
Is this safe? It allows for errors in iconType. Eg choosing one that doesn't exist. This will showup in runtime rather than compiletime.